hack/build-cluster-api.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+set -e
+
+# Check if the OPENSHIFT_INSTALL_CLUSTER_API is not empty.
+if [ -z "${OPENSHIFT_INSTALL_CLUSTER_API}" ]; then
+  return
+fi
+
+TARGET_OS_ARCH=$(go env GOOS)_$(go env GOARCH)
+CLUSTER_API_BIN_DIR="${PWD}/cluster-api/bin/${TARGET_OS_ARCH}"
+CLUSTER_API_MIRROR_DIR="${PWD}/pkg/cluster-api/mirror/"
+ENVTEST_K8S_VERSION="1.28.0"
+ENVTEST_ARCH=$(go env GOOS)-$(go env GOARCH)
+
+copy_cluster_api_to_mirror() {
+  mkdir -p "${CLUSTER_API_BIN_DIR}"
+  mkdir -p "${CLUSTER_API_MIRROR_DIR}"
+
+  # Clean the mirror, but preserve the README file.
+  rm -rf "${CLUSTER_API_MIRROR_DIR:?}/*.zip"
+
+  sync_envtest
+
+  # Zip every binary in the folder into a single zip file.
+  zip -j1 "${CLUSTER_API_MIRROR_DIR}/cluster-api.zip" "${CLUSTER_API_BIN_DIR}"/*
+}
+
+sync_envtest() {
+  if [ -f "${CLUSTER_API_BIN_DIR}/kube-apiserver" ]; then
+    version=$("${CLUSTER_API_BIN_DIR}/kube-apiserver" --version || echo "Kubernetes v0.0.0")
+    echo "Found envtest binaries with version: ${version}"
+    if [ "${version}" = "Kubernetes v${ENVTEST_K8S_VERSION}" ]; then
+      return
+    fi
+  fi
+
+  bucket="https://storage.googleapis.com/kubebuilder-tools"
```

Comment:
For this also (in addition to the providers) I think we may want to move away from either upstream or at least external dependencies (i.e. we would at least want to mirror these dependencies in our build pipeline). But this is fine for now, just checking alignment for the future.